src/gourd_lib/config/mod.rs

```diff
@@ -75,6 +74,31 @@ pub struct UserProgram {
     pub next: Vec<String>,
 }
 
+/// The user-facing side of [`Afterscript`]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Hash, Eq)]
+pub enum UserAfterscript {
```

Comment:
I actually think this makes the afterscript system more complex.

Why do we need all of these variants? The way there were handled before seemed much simpler.
Whilst I agree with some of the other changes this new struct should not be here.

Reply:
the goal here is to allow for more options in after scripts, such as specifying whether input should be passed as a file path or contents and through arguments vs stdin, maybe users explicitly want the creation of a file for example. while none of these options are implemented, in the current system we would either have them as fields in `program` which is even worse imo, or require users to specify a struct-like declaration of every afterscript. this allows for both
```
afterscript = "/path/to/script.sh"
```
and 
```
afterscript = { executable = "/path/to/script.sh", option_a = "something", option_b = true }
```
to be valid. I found this to be most friendly to new users who shouldn't be required to understand the options, and let more experienced users customise the afterscript execution however they want.

Do you believe 
a) afterscripts should be kept dead simple and any customisation is unnecessary complication that we shouldn't support anyway
b) there should be a different solution for supporting both one-line and multiple option declaration
c) we should have the complex declaration for all after scripts 
d) something else?

Reply:
a) yes
b) no
c) no
d) the users are already writing a custom `.sh` script that gourd executes, there is no need for these options as the language the shell provides can do anything, I'd focus on providing concrete examples on how to do something with an after script that are easily accessible with `gourd init` for example instead of adding new options.

Reply:
My original idea was to cater for externally provided scripts, without needing the user to modify them. To be honest I'm giving up on the idea and instead providing examples for how to deal with an external script needing the input provided differently. Simple it is!